src/query/functions/binary/common.go

```diff
@@ -107,6 +76,34 @@ func tagMap(t models.Tags) map[string]models.Tag {
 	return m
 }
 
+// Iff one of left or right is a time block, match match one to many
+// against it, and match everything.
+func defaultVectorMatcherBuilder(lhs, rhs block.Block) *VectorMatching {
```

Comment:
Can we just make `VectorMatching` be a non-ptr type?  Then it shouldn't panic if accidentally used without being set.

Reply:
Sure, was mostly using it for the unset case, will add a bool indicating that it's been updated